pkg/webhookadmission/webhook.go

```diff
@@ -94,82 +93,46 @@ func (c *Controller) processMutateRequest(w http.ResponseWriter, req *http.Reque
 	arReq := admissionReview.Request
 	resourceName := arReq.Name
 
-	switch arReq.Kind.Kind {
-	case "StatefulSet":
-		var ss appv1.StatefulSet
-		if err = json.Unmarshal(arReq.Object.Raw, &ss); err != nil {
-			log.Errorf("Could not unmarshal admission review object: %v", err)
-			c.Recorder.Event(webhookConfig, v1.EventTypeWarning, "could not unmarshal ar object", err.Error())
-			http.Error(w, "Decode error", http.StatusBadRequest)
-			return
-		}
-		if resourceName == "" {
-			resourceName = ss.GenerateName
-		}
-		log.Debugf("Received admission review request for sts %s,%s", resourceName, arReq.Namespace)
-		if !skipSchedulerUpdate(skipHookAnnotation, ss.ObjectMeta.Annotations) {
-			isStorkResource, err = c.checkVolumeOwner(ss.Spec.Template.Spec.Volumes, arReq.Namespace)
-			if err != nil {
-				c.Recorder.Event(&ss, v1.EventTypeWarning, "Could not get volume owner info for ss: %v", err.Error())
-				http.Error(w, "Could not get volume owner info", http.StatusInternalServerError)
-				return
-			}
-			schedPath = appSchedPrefix + podSpecSchedPath
-		}
-	case "Deployment":
-		var deployment appv1.Deployment
-		if err := json.Unmarshal(arReq.Object.Raw, &deployment); err != nil {
-			log.Errorf("Could not unmarshal admission review object: %v", err)
-			c.Recorder.Event(webhookConfig, v1.EventTypeWarning, "could not unmarshal ar object", err.Error())
-			http.Error(w, "Decode error", http.StatusBadRequest)
-			return
-		}
-		if resourceName == "" {
-			resourceName = deployment.GenerateName
-		}
-		log.Debugf("Received admission review request for deployment %s,%s", resourceName, arReq.Namespace)
-		if !skipSchedulerUpdate(skipHookAnnotation, deployment.ObjectMeta.Annotations) {
-			isStorkResource, err = c.checkVolumeOwner(deployment.Spec.Template.Spec.Volumes, arReq.Namespace)
-			if err != nil {
-				c.Recorder.Event(&deployment, v1.EventTypeWarning, "Could not get volume owner info deployment: %v", err.Error())
-				http.Error(w, "Could not get volume owner info", http.StatusInternalServerError)
-				return
-			}
-			schedPath = appSchedPrefix + podSpecSchedPath
-		}
-	case "Pod":
-		var pod v1.Pod
-		if err := json.Unmarshal(arReq.Object.Raw, &pod); err != nil {
-			log.Errorf("Could not unmarshal admission review object: %v", err)
-			c.Recorder.Event(webhookConfig, v1.EventTypeWarning, "could not unmarshal ar object", err.Error())
-			http.Error(w, "Decode error", http.StatusBadRequest)
+	if arReq.Kind.Kind != "Pod" {
+		errMsg := fmt.Errorf("kind=%s not supported", arReq.Kind.Kind)
+		log.Errorf("Failed to serve admission review request: %v", err)
+		c.Recorder.Event(webhookConfig, v1.EventTypeWarning, "invalid admission review request", errMsg.Error())
+		http.Error(w, err.Error(), http.StatusBadRequest)
```

Comment:
Should we just log the event on the requesting object and not return an error here? Why block a request if we cannot process it.